### ISSUE TITLE
TSQL parsing error on valid view creation select statement.

### ISSUE THREAD

**Issue by david-shanks** [sqlfluff/sqlfluff#5568]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When parsing a `create view` SQL statement with TSQL dialect, the linter throws the following error: "Found unparsable section: 'CREATE VIEW'." When I remove the DISTINCT clause from my first select statement, the parsing error goes away.

### Expected Behaviour

The linter parses the valid SQL script correctly.

### Observed Behaviour

Unable to parse a SQL script that contains a DISTINCT declaration in its select statement.

### How to reproduce

Attempt to parse any SQL script with a DISTINCT declaration in the select statement using the TSQL dialect.

### Dialect

TSql

### Version

2.3.5

### Configuration

[sqlfluff]
exclude_rules = layout.end_of_file, references.quoting, ambiguous.column_count, structure.column_order, capitalisation.identifiers

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by sgoldberg16** [comment]
My team has been having the same issue parsing DISTINCT in select statements

**Comment by FrankRua** [comment]
+1 also having this issue on my side.

**Comment by keraion** [comment]
@david-shanks Can you provide an example query where this produces the failure to parse?

**Comment by david-shanks** [comment]
> @david-shanks Can you provide an example query where this produces the failure to parse?

Here's is an example script that is failing to lint:

`CREATE VIEW [schema].[view_name] 
AS
SELECT DISTINCT TOP (100) z1.column1 AS 'exampleName' 
FROM dbo.table AS z1 WITH (NOLOCK)
ORDER BY 'exampleName'
GO`

![image](https://github.com/sqlfluff/sqlfluff/assets/40175307/dc0e28fe-9baa-4ad0-ace3-035673f5bb20)

**Comment by keraion** [comment]
This might be a duplicate of #5484 which is fixed in main and will be in the next release.

**Comment by FrankRua** [comment]
@keraion When is the next release so I can plan to re-build containers ?

**Comment by nathanthorell** [comment]
@david-shanks in the example you provided, the word table is a t-sql reserved word, so thats whats making the statement un-parsable.  It needs to be bracketed.  When you change it to:
```sql
CREATE VIEW [schema].[view_name] AS
SELECT DISTINCT TOP (100) z1.column1 AS exampleName
FROM dbo.[table] AS z1 WITH (NOLOCK)
ORDER BY exampleName GO
```
Then is parses fine.

I cannot reproduce this error in the current 3.3.0 release.  Maybe it was fixed in an earlier release as @keraion suggested and likely this ticket can be closed?
